apps/dashboard/app/javascript/packs/script_edit.js

```diff
@@ -125,11 +125,33 @@ function addInProgressField(event) {
   entireDiv.remove();
 }
 
+function fixedFieldEnabled(checkbox, dataElement) {
+  dataElement.disabled = true;
+  const input = $('<input>').attr('type','hidden').attr('name', dataElement.name).attr('value', dataElement.value);
+  $(checkbox).after(input);
+}
+
+function toggleFixed(event) {
+  const dataElement = document.getElementById(event.target.dataset.selectId);
+  if (event.target.checked) {
+    fixedFieldEnabled(event.target, dataElement)
+  } else {
+    dataElement.disabled = false;
+    $(`input[type=hidden][name="${dataElement.name}"]`).remove();
+  }
+}
+
 function enableOrDisableSelectOption(event) {
   const toggleAction = event.target.dataset.selectToggler;
   const li = event.target.parentElement;
   event.target.disabled = true;
 
+  if(toggleAction == 'fixed') {
+    toggleFixed(event)
+    event.target.disabled = false;
+    return
+  }
```

Comment:
I see this is a little bit of an optimization - but I'm not sure we should be mixing the 2 things - check boxes & add/remove buttons. 

This all maps from `data-select-toggler` - but these fixed value check boxes can apply to more than just `select` options.

I just feel like we're mixing 2 different ideas - can you refactor to separate them a little?

Reply:
That was my initial idea too... Refactoring...

Reply:
Fixed